src/file/properties.rs

```diff
@@ -0,0 +1,550 @@
+const DEFAULT_ENCODING: Encoding = Encoding::PLAIN;
+const DEFAULT_COMPRESSION: Compression = Compression::UNCOMPRESSED;
+const DEFAULT_DICTIONARY_ENABLED: bool = true;
+const DEFAULT_DICTIONARY_PAGE_SIZE_LIMIT: usize = DEFAULT_PAGE_SIZE;
+const DEFAULT_STATISTICS_ENABLED: bool = true;
+const DEFAULT_MAX_STATISTICS_SIZE: usize = 4096;
+const DEFAULT_MAX_ROW_GROUP_LENGTH: usize = 64 * 1024 * 1024;
+const DEFAULT_CREATED_BY: &str = env!("PARQUET_CREATED_BY");
+
+/// Parquet writer version.
+///
+/// Basic constant, which is not part of the Thrift definition.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum WriterVersion {
+  PARQUET_1_0,
+  PARQUET_2_0
+}
+
+impl WriterVersion {
+  /// Returns writer version as `i32`.
+  pub fn as_num(&self) -> i32 {
+    match self {
+      WriterVersion::PARQUET_1_0 => 1,
+      WriterVersion::PARQUET_2_0 => 2
+    }
+  }
+}
+
+/// Reference counted writer properties.
+pub type WriterPropertiesPtr = Rc<WriterProperties>;
+
+/// Writer properties.
+///
+/// It is created as an immutable data structure, use [`WriterPropertiesBuilder`] to
+/// assemble the properties.
+#[derive(Debug, Clone)]
+pub struct WriterProperties {
+  data_pagesize_limit: usize,
+  dictionary_pagesize_limit: usize,
+  write_batch_size: usize,
+  max_row_group_length: usize,
+  writer_version: WriterVersion,
+  created_by: String,
+  default_column_properties: ColumnProperties,
+  column_properties: HashMap<ColumnPath, ColumnProperties>
+}
+
+impl WriterProperties {
+  /// Returns builder for writer properties with default values.
+  pub fn builder() -> WriterPropertiesBuilder {
+    WriterPropertiesBuilder::with_defaults()
+  }
+
+  /// Returns data page size limit.
+  pub fn data_pagesize_limit(&self) -> usize {
+    self.data_pagesize_limit
+  }
+
+  /// Returns dictionary page size limit.
+  pub fn dictionary_pagesize_limit(&self) -> usize {
+    self.dictionary_pagesize_limit
+  }
+
+  /// Returns configured batch size for writes.
+  ///
+  /// When writing a batch of data, this setting allows to split it internally into
+  /// smaller batches so we can better estimate the size of a page currently being
+  /// written.
+  pub fn write_batch_size(&self) -> usize {
+    self.write_batch_size
+  }
+
+  /// Returns max size for a row group.
+  pub fn max_row_group_length(&self) -> usize {
+    self.max_row_group_length
+  }
+
+  /// Returns configured writer version.
+  pub fn writer_version(&self) -> WriterVersion {
+    self.writer_version
+  }
+
+  /// Returns `created_by` string.
+  pub fn created_by(&self) -> &str {
+    &self.created_by
+  }
+
+  /// Returns encoding for a data page, when dictionary encoding is enabled.
+  /// This is not configurable.
+  #[inline]
+  pub fn dictionary_data_page_encoding(&self) -> Encoding {
+    // PLAIN_DICTIONARY encoding is deprecated in writer version 1.
+    // Dictionary values are encoded using RLE_DICTIONARY encoding.
+    Encoding::RLE_DICTIONARY
```

Comment:
I do not think we should be using deprecated PLAIN_DICTIONARY. Note that parquet-cpp and parquet-mr specify them when using writer version 1.

Reply:
Sounds good to me.